params/config.go

```diff
@@ -1218,6 +1218,40 @@ func (c *ChainConfig) LatestFork(time uint64) forks.Fork {
 	}
 }
 
+// BlobConfig returns the blob config associated with the provided fork.
+func (c *ChainConfig) BlobConfig(fork forks.Fork) *BlobConfig {
+	switch fork {
+	case forks.Osaka:
+		return DefaultOsakaBlobConfig
+	case forks.Prague:
+		return DefaultPragueBlobConfig
+	case forks.Cancun:
+		return DefaultCancunBlobConfig
+	default:
+		return nil
+	}
+}
```

Comment:
While reviewing this I noticed it's not using the receiver `c` at all, and it looks like an upstream config bug.
It was fixed here, to use the actual config contents: https://github.com/ethereum/go-ethereum/pull/32579
But that fix wasn't included in the 1.16.3 release, so it will use the default configs instead here (!!!).
We should be aware of this difference; usage of the blob config won't be accurate on custom BPO settings.

Reply:
Great catch. I'll file an issue in this repo for that and reference in a code comment here. We expect to do another upstream merge very shortly which will rectify this problem.